Makefile

```diff
@@ -183,5 +182,379 @@ verify_version:
+	$(Q) $(MAKE) test TESTS=tests/Integrations/Lumen/V5_8
+test_web_slim_312:
+	$(Q) $(COMPOSER) --working-dir=tests/Frameworks/Slim/Version_3_12 update
+	$(Q) $(MAKE) test TESTS=--testsuite=slim-312-test
+test_web_symfony_23:
+	$(Q) $(COMPOSER) --working-dir=tests/Frameworks/Symfony/Version_2_3 update
+	$(Q) $(MAKE) test TESTS=tests/Integrations/Symfony/V2_3
+test_web_symfony_28:
+	$(Q) $(COMPOSER) --working-dir=tests/Frameworks/Symfony/Version_2_8 update
+	$(Q) $(MAKE) test TESTS=tests/Integrations/Symfony/V2_8
+test_web_symfony_30:
+	$(Q) $(COMPOSER) --working-dir=tests/Frameworks/Symfony/Version_3_0 update
+	$(Q) php tests/Frameworks/Symfony/Version_3_0/bin/console cache:clear --no-warmup --env=prod
+	$(Q) $(MAKE) test TESTS=tests/Integrations/Symfony/V3_0
+test_web_symfony_33:
+	$(Q) $(COMPOSER) --working-dir=tests/Frameworks/Symfony/Version_3_3 update
+	$(Q) php tests/Frameworks/Symfony/Version_3_3/bin/console cache:clear --no-warmup --env=prod
+	$(Q) $(MAKE) test TESTS=tests/Integrations/Symfony/V3_3
+test_web_symfony_34:
+	$(Q) $(COMPOSER) --working-dir=tests/Frameworks/Symfony/Version_3_4 update
+	$(Q) php tests/Frameworks/Symfony/Version_3_4/bin/console cache:clear --no-warmup --env=prod
+	$(Q) $(MAKE) test TESTS=tests/Integrations/Symfony/V3_4
+test_web_symfony_40:
+	$(Q) $(COMPOSER) --working-dir=tests/Frameworks/Symfony/Version_4_0 update
+	$(Q) php tests/Frameworks/Symfony/Version_4_0/bin/console cache:clear --no-warmup --env=prod
+	$(Q) $(MAKE) test TESTS=tests/Integrations/Symfony/V4_0
+test_web_symfony_42:
+	$(Q) $(COMPOSER) --working-dir=tests/Frameworks/Symfony/Version_4_2 update
+	$(Q) php tests/Frameworks/Symfony/Version_4_2/bin/console cache:clear --no-warmup --env=prod
+	$(Q) $(MAKE) test TESTS=tests/Integrations/Symfony/V4_2
+test_web_symfony_44:
+	$(Q) $(COMPOSER) --working-dir=tests/Frameworks/Symfony/Version_4_4 update
+	$(Q) php tests/Frameworks/Symfony/Version_4_4/bin/console cache:clear --no-warmup --env=prod
+	$(Q) $(MAKE) test TESTS=tests/Integrations/Symfony/V4_4
+test_web_wordpress_48:
+	$(Q) $(MAKE) test TESTS=tests/Integrations/WordPress/V4_8
+test_web_yii_2:
+	$(Q) $(COMPOSER) --working-dir=tests/Frameworks/Yii/Version_2_0_26 update
+	$(Q) $(MAKE) test TESTS=tests/Integrations/Yii/V2_0_26
+test_web_zend_1:
+	$(Q) $(MAKE) test TESTS=tests/Integrations/ZendFramework/V1
+test_web_custom:
+	$(Q) $(COMPOSER) --working-dir=tests/Frameworks/Custom/Version_Autoloaded update
+	$(Q) $(MAKE) test TESTS=--testsuite=custom-framework-autoloaded-test
+
+
+test_scenario_%: $(TESTS_ROOT)/composer.lock
+	$(Q) $(COMPOSER) --working-dir=$(TESTS_ROOT) scenario $*
+
+$(TESTS_ROOT)/composer.lock: $(TESTS_ROOT)/composer.json
+	$(Q) $(COMPOSER) update
+
+### Api tests ###
+API_TESTS_ROOT := ./tests_api
+
+test_api_unit: composer.lock
+	$(Q) $(ENV_OVERRIDE) php $(REQUEST_INIT_HOOK) vendor/bin/phpunit --config=phpunit.xml $(API_TESTS_ROOT)/Unit $(TESTS)
+
+composer.lock: composer.json
+	$(Q) composer update
+
```

Comment:
Nice idea using make! This might already be on your TODO list: could we also move `composer compile` here too? Something like `make generated` or something? That way we won't need composer to make the generated file.

Reply:
We need composer to make the generated file because of the dev dependencies -- so it's not particularly helpful to avoid composer for that specific usage.

Reply:
Ah yes, good point.

Reply:
So if you agree I leave the compilation in composer, but we will hopefully get rid of it if decide to include the required PHP files in the binary.